src/script/lua_api/l_object.cpp

```diff
@@ -950,59 +859,61 @@ int ObjectRef::l_get_acceleration(lua_State *L)
 {
 	NO_MAP_LOCK_REQUIRED;
 	ObjectRef *ref = checkobject(L, 1);
-	LuaEntitySAO *co = getluaobject(ref);
-	if (co == NULL) return 0;
-	// Do it
-	v3f v = co->getAcceleration();
-	pushFloatPos(L, v);
+	LuaEntitySAO *entitysao = getluaobject(ref);
+	if (entitysao == nullptr)
+		return 0;
+
+	v3f acceleration = entitysao->getAcceleration();
+	pushFloatPos(L, acceleration);
 	return 1;
 }
 
-// set_rotation(self, {x=num, y=num, z=num})
-// Each 'num' is in radians
+// set_rotation(self, rotation)
 int ObjectRef::l_set_rotation(lua_State *L)
 {
 	NO_MAP_LOCK_REQUIRED;
 	ObjectRef *ref = checkobject(L, 1);
-	LuaEntitySAO *co = getluaobject(ref);
-	if (!co)
+	LuaEntitySAO *entitysao = getluaobject(ref);
+	if (entitysao == nullptr)
 		return 0;
 
 	v3f rotation = check_v3f(L, 2) * core::RADTODEG;
-	co->setRotation(rotation);
+
+	entitysao->setRotation(rotation);
 	return 0;
 }
 
 // get_rotation(self)
-// returns: {x=num, y=num, z=num}
-// Each 'num' is in radians
 int ObjectRef::l_get_rotation(lua_State *L)
 {
 	NO_MAP_LOCK_REQUIRED;
 	ObjectRef *ref = checkobject(L, 1);
-	LuaEntitySAO *co = getluaobject(ref);
-	if (!co)
+	LuaEntitySAO *entitysao = getluaobject(ref);
+	if (entitysao == nullptr)
 		return 0;
 
+	v3f rotation = entitysao->getRotation() * core::DEGTORAD;
+
 	lua_newtable(L);
-	v3f rotation = co->getRotation() * core::DEGTORAD;
 	push_v3f(L, rotation);
 	return 1;
 }
 
-// set_yaw(self, radians)
+// set_yaw(self, yaw)
 int ObjectRef::l_set_yaw(lua_State *L)
 {
 	NO_MAP_LOCK_REQUIRED;
 	ObjectRef *ref = checkobject(L, 1);
-	LuaEntitySAO *co = getluaobject(ref);
+	LuaEntitySAO *entitysao = getluaobject(ref);
+	if (entitysao == nullptr)
+		return 0;
 
-	if (co == NULL) return 0;
 	if (isNaN(L, 2))
 		throw LuaError("ObjectRef::set_yaw: NaN value is not allowed.");
 
 	float yaw = readParam<float>(L, 2) * core::RADTODEG;
-	co->setRotation(v3f(0, yaw, 0));
+
+	entitysao->setRotation(v3f(0, yaw, 0));
```

Comment:
Question (not related to this cleanup, just a general question), but doesn't setting the other two to `0` reset the pitch and roll of `ObjectRef:set_rotation`?  If that's so, sounds like a bug.

Reply:
I have a feeling that doing this any other way may mess with XYZ/ZYX ordering - ie: the yaw won't turn out correctly